internal/datastreams/pathway.go

```diff
@@ -17,13 +17,24 @@ import (
 
 var hashableEdgeTags = map[string]struct{}{"event_type": {}, "exchange": {}, "group": {}, "topic": {}, "type": {}, "direction": {}}
 
+func isValidArn(arn string) bool {
+	parts := strings.Split(arn, ":")
+	if len(parts) > 0 && parts[0] == "arn" && len(parts) > 5 {
+		return true
+	}
+	return false
```

Comment:
I'd suggest a different approach that is faster:

```suggestion
	separators := strings.Count(arn, ":")
	if separators < 5 {
		return false
	}
	if strings.HasPrefix(arn, "arn:") {
		return true
	}
	return false
```

I wrote a benchmark to validate it. This is the original `isValidArn` results:

```
BenchmarkIsValidArn-10    	10121137	       110.5 ns/op	     112 B/op	       1 allocs/op
```

And this is the results for the suggested code:

```
BenchmarkIsValidArn-10    	53713204	        20.14 ns/op	       0 B/op	       0 allocs/op
```

Reply:
Made the change! Thank you!